storage/client_test.go

```diff
@@ -1415,6 +1416,43 @@ func TestRetryMaxAttemptsEmulated(t *testing.T) {
 	})
 }
 
+// Test that a timeout returns a DeadlineExceeded error, in spite of DeadlineExceeded being a retryable
+// status when it is returned by the server.
+func TestTimeoutErrorEmulated(t *testing.T) {
+	transportClientTest(t, func(t *testing.T, project, bucket string, client storageClient) {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Nanosecond)
```

Comment:
Confused by this test, is this to cause DeadlineExceeded client-side while the other is meant for server side (storage-testbench) to cause DeadlineExceeded?

Reply:
Yes exactly. We want to make sure that client-set deadlines are not retried (to avoid issue mentioned in https://google.aip.dev/194#non-retryable-codes ) but that a DEADLINE_EXCEEDED returned by the server is retried.